src/index.ts

```diff
@@ -73,22 +70,18 @@ export default function createClient<T>(defaultOptions?: ClientOptions) {
 
   /** Gets a union of paths which have method */
   type PathsWith<M extends Method> = {
-    [Path in keyof T]: T[Path] extends { [ K in M ]: unknown } ? Path : never
+    [Path in keyof T]: T[Path] extends { [K in M]: unknown } ? Path : never;
   }[keyof T];
 
-  type PathParams<U extends keyof T> = T[U] extends { parameters: any } ? { params: T[U]['parameters'] } : { params?: BaseParams };
-  type MethodParams<U extends keyof T, M extends keyof T[U]> = T[U][M] extends {
+  type PathParams<T> = T extends { parameters: any } ? { params: T['parameters'] } : { params?: BaseParams };
+  type MethodParams<T> = T extends {
     parameters: any;
   }
-    ? { params: T[U][M]['parameters'] }
+    ? { params: T['parameters'] }
     : { params?: BaseParams };
-  type Params<U extends keyof T, M extends keyof T[U]> = PathParams<U> & MethodParams<U, M>;
```

Comment:
Oddly enough, this cleanup made debugging much easier. Passing the same nested generics through inference functions not only confuses the dev; the compiler gets lost too. The nested generics are moved to only appear on `FetchOptions<>` to clean this up.